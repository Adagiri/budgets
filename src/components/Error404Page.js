import React from "react";
import { Link } from "react-router-dom";

const Error404Page = props => (
  <div>
    <h1>
      404! <Link to="/">Go home</Link>
    </h1>
  </div>
);

export default Error404Page;
