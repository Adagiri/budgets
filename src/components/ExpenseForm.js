import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";

export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.updateDescription = this.updateDescription.bind(this);
    this.updateAmount = this.updateAmount.bind(this);
    this.updateNote = this.updateNote.bind(this);
    this.submit = this.submit.bind(this);
    this.onDateChange = this.onDateChange.bind(this);
    this.onFocusChange = this.onFocusChange.bind(this);
    this.state = {
      description: props.expense ? props.expense.description : "",
      amount: props.expense ? (props.expense.amount / 100).toString() : "",
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      note: props.expense ? props.expense.note : "",
      error: "",
      focused: false
    };
  }

  submit = e => {
    e.preventDefault();
    !this.state.description || !this.state.amount
      ? this.setState(() => ({ error: "Please add description and amount" }))
      : this.props.onSubmit({
          description: this.state.description,
          amount: parseFloat(this.state.amount, 10) * 100,
          note: this.state.note,
          createdAt: this.state.createdAt.valueOf()
        });
  };

  updateDescription = e => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };

  updateAmount = e => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };

  updateNote = e => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };

  onDateChange = createdAt => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };

  onFocusChange = ({ focused }) => this.setState(() => ({ focused }));

  render() {
    return (
      <div>
        <h4>{this.state.error}</h4>
        <form onSubmit={this.submit}>
          <input
            type="text"
            placeholder="Description"
            value={this.state.description}
            onChange={this.updateDescription}
            autoFocus
          />

          <input
            type="text"
            placeholder="Amount"
            value={this.state.amount}
            onChange={this.updateAmount}
          />

          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.focused}
            onFocusChange={this.onFocusChange}
            isOutsideRange={() => false}
            numberOfMonths={1}
          />

          <textarea
            placeholder="Add a note for your expense(optional)"
            value={this.state.note}
            onChange={this.updateNote}
          ></textarea>
          <button>Add expense</button>
        </form>
      </div>
    );
  }
}
