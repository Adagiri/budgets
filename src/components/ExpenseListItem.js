import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

const ExpenseListItem = ({ id, amount, createdAt, description }) => (

    <Link className="list-item" to={`/edit/${id}`}>
    <div>
    <h3> Description: {description}</h3>
    {moment(createdAt).format("MMMM Do, YYYY")}
    </div>
      <h3>
      {numeral(amount / 100).format("$0,0.00")}
    </h3>
    </Link>
  
);

export default ExpenseListItem;
