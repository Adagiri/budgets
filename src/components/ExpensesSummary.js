import React from "react";
import { Link } from 'react-router-dom';
import ExpenseTotal from "../selectors/ExpensesTotal";
import { connect } from "react-redux";
import getVisibleExpenses from "../selectors/getVisibleExpenses";
import numeral from "numeral";

const ExpensesSummary = props => (
  <div className="page-header">
  <div className="content-container">
  <h1 className="page-header__title">
    Viewing <span>{props.expenses.length}</span> expenses totalling{" "}
    <span>{numeral(ExpenseTotal(props.expenses) / 100).format("$0,0.00")}</span>
  </h1>
    {
      props.unfiltered.length > props.expenses.length ? <h3 className="page-header__title not-viewing"> Not viewing <span>{props.unfiltered.length - props.expenses.length}</span> expense because of filters</h3> : ""
    }
    <div className="page-header__action">
    <Link className="button" to="/create">Add Expense</Link>
    </div>
  </div>
  </div>
);

const mapStateToProps = state => {
  return {
    unfiltered: state.expenses,
    expenses: getVisibleExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
