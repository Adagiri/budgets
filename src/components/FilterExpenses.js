import React from "react";
import { connect } from "react-redux";
import "react-dates/lib/css/_datepicker.css";
import "react-dates/initialize";
import moment from "moment";
import {
  DateRangePicker,
  SingleDatePicker,
  DayPickerRangeController
} from "react-dates";
import {
  filterText,
  sortByAmount,
  sortByDate,
  setStartDate,
  setEndDate
} from "../actions/Filters";


class FilterExpense extends React.Component {
  state = {
    focused: null
  };

  textChange = e => {
    const text = e.target.value;
    this.props.dispatch(filterText(text));
  };

  optionChange = e => {
    const option = e.target.value;
    option === "date"
      ? this.props.dispatch(sortByDate())
      : this.props.dispatch(sortByAmount());
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
    console.log(moment(startDate)._d);
  };
  // onDatesChange = ({ startDate, endDate }) => {
  //    this.props.setStartDate(startDate);
  //    this.props.setEndDate(endDate);
  //  };

  onFocusChange = focused => {
    this.setState(() => ({ focused }));
  };

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.textChange}
        />
        <select onChange={this.optionChange}>
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>

        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          startDateId="startDate"
          endDateId="endDate"
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.focused}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
          displayFormat={() => "DD/MM/YYYY"}
        />
      </div>
    );
  }
}

const MapStateWithProps = state => ({ filters: state.filters });

export default connect(MapStateWithProps)(FilterExpense);
