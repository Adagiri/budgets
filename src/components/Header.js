import React from "react";
import { NavLink, Link } from "react-router-dom";
import { startLogout } from "../actions/auth";
import { connect } from 'react-redux';

export const Header = ({ startLogout }) => (
  <div>
    <h1>Expensify</h1>
    <NavLink to="/dashboard">Dashboard</NavLink>
    <NavLink to="/create">Add Expense</NavLink>
    <button onClick={startLogout}>Logout</button>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
