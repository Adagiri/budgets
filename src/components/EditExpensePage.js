import React from "react";
import ExpenseForm from "./ExpenseForm";
import { connect } from "react-redux";
import { startRemoveExpense, startEditExpense } from "../actions/Expenses";

export class EditExpensePage extends React.Component {
  onSubmit = expense => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push("/dashboard");
  };

  onClick = () => {
    this.props.startRemoveExpense(this.props.expense.id);
    this.props.history.push("/dashboard");
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
        <div className="content-container">
          <button className="button button__secondary" onClick={this.onClick}>Remove Expense</button>
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch, props) => {
  return {
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: id => dispatch(startRemoveExpense(id))
  };
};

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id)
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
