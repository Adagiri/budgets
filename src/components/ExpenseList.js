import React from "react";
import ExpenseListItem from "./ExpenseListItem";
import { connect } from "react-redux";
import getVisibleExpenses from "../selectors/getVisibleExpenses";

export const ExpenseList = (props) =>(
    <div className="content-container">
    <div className="list-header">
    <div className="show-for-mobile">Expenses</div>
    <div className="show-for-desktop">Expense</div>    
    <div className="show-for-desktop">Amount</div> 
    </div>
    {props.expenses.map((expense) => {
      return <ExpenseListItem key={expense.id} {...expense} />;
    })
  }
    </div>
);

const ConnectStateWithProps = connect(state => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters)
  };
})(ExpenseList);

export default ConnectStateWithProps;
