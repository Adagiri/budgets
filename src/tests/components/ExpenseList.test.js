import React from 'react';
import ExpenseArray from '../fixtures/expenses';
import { ExpenseList }from '../../components/ExpenseList';
import { shallow } from 'enzyme';

test('should map out and view expense in expensesArray', () => {
    const wrapper = shallow( <ExpenseList expenses={ExpenseArray} />);
    expect(wrapper).toMatchSnapshot();
});