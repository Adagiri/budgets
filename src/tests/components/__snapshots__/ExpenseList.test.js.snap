// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should map out and view expense in expensesArray 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <ExpenseListItem
      amount={4500}
      createdAt={"1969-12-23T00:00:00.000Z"}
      description="Rent"
      id={1}
      key="1"
      note="My house rent"
    />
    <ExpenseListItem
      amount={3500}
      createdAt={"1970-01-01T00:00:00.000Z"}
      description="Water"
      id={2}
      key="2"
      note="Water usage rate"
    />
    <ExpenseListItem
      amount={85000}
      createdAt={"1970-01-10T00:00:00.000Z"}
      description="Fees"
      id={3}
      key="3"
      note="Outstanding fee"
    />
  </div>
</div>
`;
